idaes_examples/notebooks/docs/unit_models/custom_unit_models/creating_unit_model.ipynb

```diff
@@ -0,0 +1,2813 @@
+    "            for k in self.keys():\n",
+    "                if degrees_of_freedom(self[k]) != 0:\n",
+    "                    raise Exception(\n",
+    "                        \"State vars fixed but degrees of freedom \"\n",
+    "                        \"for state block is not zero during \"\n",
+    "                        \"initialization.\"\n",
+    "                    )\n",
+    "\n",
+    "        if state_vars_fixed is False:\n",
+    "            if hold_state is True:\n",
+    "                return flags\n",
+    "            else:\n",
+    "                self.release_state(flags)\n",
+    "\n",
+    "        init_log.info(\"Initialization Complete.\")\n",
+    "\n",
+    "    def release_state(self, flags, outlvl=idaeslog.NOTSET):\n",
+    "        \"\"\"\n",
+    "        Method to release state variables fixed during initialization.\n",
+    "        Keyword Arguments:\n",
+    "            flags : dict containing information of which state variables\n",
+    "                    were fixed during initialization, and should now be\n",
+    "                    unfixed. This dict is returned by initialize if\n",
+    "                    hold_state=True.\n",
+    "            outlvl : sets output level of logging\n",
+    "        \"\"\"\n",
+    "        init_log = idaeslog.getInitLogger(self.name, outlvl, tag=\"properties\")\n",
+    "\n",
+    "        if flags is None:\n",
+    "            return\n",
+    "        # Unfix state variables\n",
+    "        revert_state_vars(self, flags)\n",
+    "        init_log.info(\"State Released.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1f34c3ff",
+   "metadata": {},
+   "source": [
+    "The class `OrgPhaseStateBlockData` is designated with the `declare_process_block_class` decorator, named `OrgPhaseStateBlock`, and inherits the block class from `_StateBlock`. This inheritance allows `OrgPhaseStateBlockData` to leverage functions from `_StateBlock`. Following the class definition, a build function similar to the one used in the `PhysicalParameterData` block is employed. The super function is utilized to enable the utilization of functions from the parent or sibling class.\n",
+    "\n",
+    "The subsequent objective is to delineate the state variables, accomplished through the `_make_state_vars` method. This method encompasses all the essential state variables and associated data. For this particular property package, the required state variables are:\n",
+    "\n",
+    "- `flow_vol` - volumetric flow rate\n",
+    "- `conc_mass_comp` - mass fractions\n",
+    "- `pressure` - state pressure\n",
+    "- `temperature` - state temperature\n",
+    "\n",
+    "\n",
+    "After establishing the state variables, the subsequent step involves setting up state properties as constraints. This includes specifying the relationships and limitations that dictate the system's behavior. The following properties need to be articulated:\n",
```

Comment:
I think some of these are missing underscores, check that they match the variable names.